From 3081b22fe077da62c132cfa058197495871d6714 Mon Sep 17 00:00:00 2001
Message-Id: <3081b22fe077da62c132cfa058197495871d6714.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Tue, 13 Jan 2015 22:32:20 +0530
Subject: [PATCH 45/95] MUST_REBASE [VPG]: drivers/video/adf: MIPI Timings
 related changes for dual link

hactive, hfp, hbp, hsync needs to be halved for dual link MIPI Panels.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

For: GMINL-5276
Change-Id: Ia555a7b4a22c0205e03a875fd100bf9203c1ea3d
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_dsi_port.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_dsi_port.c b/drivers/video/adf/intel/core/vlv/vlv_dsi_port.c
index 8849a79..2f4e685 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dsi_port.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dsi_port.c
@@ -139,6 +139,15 @@ static void set_dsi_timings(struct vlv_dsi_port *port,
 	hsync = mode->hsync_end - mode->hsync_start;
 	hbp = mode->htotal - mode->hsync_end;
 
+	if (intel_dsi->dual_link) {
+		hactive /= 2;
+		if (intel_dsi->dual_link == DSI_DUAL_LINK_FRONT_BACK)
+			hactive += intel_dsi->pixel_overlap;
+		hfp /= 2;
+		hsync /= 2;
+		hbp /= 2;
+	}
+
 	vfp = mode->vsync_start - mode->vdisplay;
 	vsync = mode->vsync_end - mode->vsync_start;
 	vbp = mode->vtotal - mode->vsync_end;
-- 
1.7.9.5

